main.js

```diff
@@ -45,10 +45,76 @@ for (let i = 0; i < arr9.length; i++) {
 }
 
 //task 10
-let arr10 = [2, 5, 8, 15, 0, 6, 20, 3];
+const arr10 = [2, 5, 8, 15, 0, 6, 20, 3];
 
 for (let i = 0; i < arr10.length; i++) {
     if (arr10[i] % 2 === 0) {
         console.log(arr10[i]);
     }
-}
\ No newline at end of file
+}
+
+//task 18.5
+const newFormatData = function (dataFun) {
+    return dataFun.replace(/(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})/g, '$<day>.$<month>.$<year>');
+}
+
+const dataF = '2020-11-26'
+console.log(newFormatData(dataF));
+
+
+//task 18.6
+const data = [
+    {
+        country: 'Russia',
+        city: 'Saint Petersburg',
+        hotel: 'Hotel Leopold',
+    },
+    {
+        country: 'Spain',
+        city: 'Santa Cruz de Tenerife',
+        hotel: 'Apartment Sunshine',
+    },
+    {
+        country: 'Slowakia',
+        city: 'Vysokie Tatry',
+        hotel: 'Villa Kunerad',
+    },
+    {
+        country: 'Germany',
+        city: 'Berlin',
+        hotel: 'Hostel Friendship',
+    },
+    {
+        country: 'Indonesia',
+        city: 'Bali',
+        hotel: 'Ubud Bali Resort&SPA',
+    },
+    {
+        country: 'Netherlands',
+        city: 'Rotterdam',
+        hotel: 'King Kong Hostel',
+    },
+    {
+        country: 'Marocco',
+        city: 'Ourika',
+        hotel: 'Rokoko Hotel',
+    },
+    {
+        country: 'Germany',
+        city: 'Berlin',
+        hotel: 'Hotel Rehberge Berlin Mitte',
+    },
+];
+
+function seachByWorld(world = '') {
+    for (let i = 0; i < data.length; i++) {
+        result = data[i].country + ', ' + data[i].city + ', ' + data[i].hotel;
+        if (data[i].city === world || data[i].country === world || data[i].hotel === world) {
+            return result;
+        }
+    }
+}
```

Comment:
It doesn't return the list of strings as required for the task.
Also, make sure the function doesn't return undefined.

Reply:
Yes, I made this function by task!